File: test-suite/src/test/java/io/micronaut/docs/http/server/bind/annotation/ShoppingCartController.java
package io.micronaut.docs.http.server.bind.annotation;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

@Controller("/customBinding")
public class ShoppingCartController {

    // tag::method[]
    @Get("/annotated")
    HttpResponse<String> checkSession(@ShoppingCart Long sessionId) { //<1>
        return HttpResponse.ok("Session:" + sessionId);
    }
    // end::method
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/server/bind/annotation/ShoppingCartController.kt
package io.micronaut.docs.http.server.bind.annotation

import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import java.util.*

@Controller("/customBinding")
class ShoppingCartController {

    // tag::method[]
    @Get("/annotated")
    fun checkSession(@ShoppingCart sessionId: Long): HttpResponse<String> { //<1>
        return HttpResponse.ok("Session:$sessionId")
    }
    // end::method[]
}
